File: 0055-jump-game/0055-jump-game.ts
function canJump(nums: number[]): boolean {
    let target: number = nums.length - 1;

    for (let i: number = nums.length - 1; i >= 0; i--) {
        if (nums[i] + i >= target) {
            target = i
        }
    }
    return target == 0
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        max_distance = 0

        for i in range(len(nums)):
            if i > max_distance:
                return False
            curr_reach = i + nums[i]
            max_distance = max(curr_reach, max_distance)
            if max_distance >= len(nums):
                return True

